examples/consumer/main.go

```
@@ -145,25 +136,32 @@ RETRY:
 
 }
 
-func createSubscription(resourceAddress string) []byte {
+func createSubscription(resourceAddress string) (sub pubsub.PubSub, err error) {
+	var status int
 	subURL := &types.URI{URL: url.URL{Scheme: "http",
 		Host: apiAddr,
 		Path: fmt.Sprintf("%s%s", apiPath, "subscriptions")}}
 	endpointURL := &types.URI{URL: url.URL{Scheme: "http",
 		Host: localAPIAddr,
 		Path: fmt.Sprintf("%s", "event")}}
 
-	pub := v1pubsub.NewPubSub(endpointURL, resourceAddress)
-	if b, err := json.Marshal(&pub); err == nil {
+	sub = v1pubsub.NewPubSub(endpointURL, resourceAddress)
+	var subB []byte
+
+	if subB, err = json.Marshal(&sub); err == nil {
 		rc := restclient.New()
-		if status, b := rc.PostWithReturn(subURL, b); status == http.StatusCreated {
-			return b
+		if status, subB = rc.PostWithReturn(subURL, subB); status != http.StatusCreated {
+			err = fmt.Errorf("subscription creation api at %s, returned status %d", subURL, status)
+			return
 		}
-		log.Errorf("subscription create returned error %s", string(b))
 	} else {
-		log.Errorf("failed to create subscription ")
+		log.Error("failed to marshal subscription ")
 	}
-	return nil
+	if err = json.Unmarshal(subB, &sub); err != nil {
+		return
```

unmarshal error is not returned here.